demos/mqtt/mqtt_demo_lightweight/mqtt_demo_lightweight.c

```diff
@@ -0,0 +1,1045 @@
+/*-----------------------------------------------------------*/
+
+static void mqttKeepAlive( int tcpSocket,
+                           MQTTFixedBuffer_t * pFixedBuffer )
+{
+    MQTTStatus_t result;
+    int status;
+    size_t packetSize = 0;
+
+    status = MQTT_GetPingreqPacketSize( &packetSize );
+
+    /* PingReq is fixed length packet, therefore there is no need to calculate the size,
+     * just makes sure static buffer can accommodate ping request. */
+    assert( packetSize <= SHARED_BUFFER_SIZE );
+
+    result = MQTT_SerializePingreq( pFixedBuffer );
+    assert( result == MQTTSuccess );
+
+    /* Send Ping Request to the broker. */
+    status = send( tcpSocket, ( void * ) pFixedBuffer->pBuffer, packetSize, 0 );
+    assert( status == ( int ) packetSize );
+}
+
+/*-----------------------------------------------------------*/
+
+static void mqttDisconnect( int tcpSocket,
+                            MQTTFixedBuffer_t * pFixedBuffer )
+{
+    MQTTStatus_t result;
+    int32_t status;
+    size_t packetSize = 0;
+
+    status = MQTT_GetDisconnectPacketSize( &packetSize );
+
+    assert( packetSize <= pFixedBuffer->size );
+
+    result = MQTT_SerializeDisconnect( pFixedBuffer );
+    assert( result == MQTTSuccess );
+
+    /* Send disconnect packet to the broker */
+    status = send( tcpSocket, ( void * ) pFixedBuffer->pBuffer, packetSize, 0 );
+    assert( status == ( int ) packetSize );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Process a response or ack to an MQTT request (PING, SUBSCRIBE
+ * or UNSUBSCRIBE). This function processes PING_RESP, SUB_ACK, UNSUB_ACK
+ *
+ * @param[in] pxIncomingPacket is a pointer to structure containing deserialized
+ * MQTT response.
+ *
+ * @param[in] packetId is packet identifier from the incoming if it was received.
+ * @note Not all responses contain packet identifier.
+ */
+static void mqttProcessResponse( MQTTPacketInfo_t * pIncomingPacket,
+                                 uint16_t packetId )
+{
+    switch( pIncomingPacket->type & 0xf0 )
+    {
+        case MQTT_PACKET_TYPE_SUBACK:
+            LogInfo( ( "Subscribed to the topic %s.\r\n", MQTT_EXAMPLE_TOPIC ) );
+            /* Make sure ACK packet identifier matches with Request packet identifier. */
+            assert( subscribePacketIdentifier == packetId );
+            break;
+
+        case MQTT_PACKET_TYPE_UNSUBACK:
+            LogInfo( ( "Unsubscribed from the topic %s.\r\n", MQTT_EXAMPLE_TOPIC ) );
+            /* Make sure ACK packet identifier matches with Request packet identifier. */
+            assert( unsubscribePacketIdentifier == packetId );
+            break;
+
+        case MQTT_PACKET_TYPE_PINGRESP:
+            LogInfo( ( "Ping Response successfully received.\r\n" ) );
+            break;
+
+        /* Any other packet type is invalid. */
+        default:
+            LogInfo( ( "mqttProcessResponse() called with unknown packet type:(%lu).", pIncomingPacket->type ) );
```

Comment:
I think `pIncomingPacket->type` should be cast to an `unsigned long` to avoid compiler warnings (unless we're not using C90 for the demos)

Reply:
Typecasted to unsigned and changed to %u since it is uint8_t, I think that is better.